plugins/woocommerce/src/Admin/Features/ProductBlockEditor/ProductTemplates/ProductBlock.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Automattic\WooCommerce\Admin\Features\ProductBlockEditor\ProductTemplates;
+
+use Automattic\WooCommerce\Admin\BlockTemplates\BlockInterface;
+use Automattic\WooCommerce\Admin\BlockTemplates\ContainerInterface;
+use Automattic\WooCommerce\Internal\Admin\BlockTemplates\Block;
+
+class ProductBlock extends Block implements ContainerInterface {
+
+	/**
+	 * Adds block to the section block.
+	 *
+	 * @param array $block_config The block data.
+	 */
+	public function &add_block(array $block_config): BlockInterface
+	{
+		$block = new ProductBlock($block_config, $this->get_root_template(), $this);
+		return $this->add_inner_block($block);
+	}
+
+	/**
+	 * Adds block to the section block.
+	 *
+	 * @param array $conditional_attributes Conditional attributes.
+	 * @param array $block_configs The nested block data.
+	 */
+	public function add_conditional_block(array $conditional_attributes, array ...$block_configs): BlockInterface
```

Comment:
A couple quick thoughts here:

1. Do we need a dedicated function to adding a conditional?  Could we just use `add_block` to add the conditional and then use its `add_block` method?
2. The API for this is not 100% established and there's still some potential for using client-side visibility tools.  See https://github.com/woocommerce/woocommerce/pull/39517

I think whether or not we go with server-side or client-side tooling for this, I think a pattern where all blocks have a method `hide_if( $condition_args )` that then gets added to the template to parse when calling `get_formatted_template` might be a better way to handle this.  Otherwise we could end up with very deeply nested conditional blocks since other template consumers won't have awareness of other conditions.

But I think saving this for a follow-up PR would be helpful so we can discuss in isolation and review outstanding discussions with the approach here.

Reply:
> Do we need a dedicated function to adding a conditional? Could we just use add_block to add the conditional and then use its add_block method?

Certainly not necessary, but it does cause some extra repetition and boilerplate, which was my reasoning for adding it, also felled easier to read this way.

> The API for this is not 100% established and there's still some potential for using client-side visibility tools.

Aaah interesting, will leave some thoughts on that PR.

> I think whether or not we go with server-side or client-side tooling for this, I think a pattern where all blocks have a method hide_if( $condition_args ) that then gets added to the template to parse when calling get_formatted_template might be a better way to handle this.

Yeah that makes sense, I like that idea.
For now I can remove this function and do the more manual approach, by just using `add_block` to wrap the conditionals. We can clean it up in a follow up PR as you mention.